olive/engine/engine.py

```diff
@@ -112,22 +109,29 @@ def __init__(
                     f"from given execution providers {self.execution_providers}."
                 )
                 accelerators = inferred_accelerators
+        logger.debug(f"Initial accelerators: {accelerators}")
 
         ep_to_process = set(self.execution_providers)
         self.accelerator_specs: List[AcceleratorSpec] = []
         is_cpu_available = "cpu" in [accelerator.lower() for accelerator in accelerators]
         for accelerator in accelerators:
             device = Device(accelerator.lower())
-            skip_get_available_ep = False
-            if (
-                self.target.system_type in (SystemType.AzureML, SystemType.Docker, SystemType.PythonEnvironment)
-                and self.target.olive_managed_env
-            ):
-                skip_get_available_ep = True
-            supported_eps = AcceleratorLookup.get_execution_providers_for_device(device, skip_get_available_ep)
+            if self.target.olive_managed_env:
+                # get all the execution providers supported by the device
+                supported_eps = AcceleratorLookup.get_execution_providers_for_device(device, skip_get_available_ep=True)
```

Comment:
>get_execution_providers_for_device [](http://example.com/codeflow?start=50&length=34)

better use a separated method to query EP for multiple EP scenario since get_execution_providers_for_device is called in perf-tuning and model preparation as well

Reply:
This changed doesn't exist anymore since I rebased to main after your PR.